### PR TITLE
Aj/disco 479 publisher orgs

### PR DESCRIPTION
**Pull request by staubina**
Updating the organizations endpoint to respect a User's permissions.

### CODE REVIEW

File: course_discovery/apps/api/v1/tests/test_views/test_organizations.py [hunk 1]
```diff
@@ -65,6 +69,35 @@ def test_list(self):
...
+        OrganizationFactory.create(partner=self.partner)
+
+        extension1 = publisher_factories.OrganizationExtensionFactory(organization=org1)
+        publisher_factories.OrganizationExtensionFactory(organization=org2)
+        assign_perm(OrganizationExtension.VIEW_COURSE, extension1.group, extension1)
+
+        self.non_staff_user.groups.add(extension1.group)
+
+        # Check Staff user get all groups
+        with self.assertNumQueries(7):
```
Comment (jlajoie) [inline course_discovery/apps/api/v1/tests/test_views/test_organizations.py]: nit: I hate numQuery checks, what is it we actually care about from these queries?

File: course_discovery/apps/api/v1/tests/test_views/test_organizations.py [hunk 2]
```diff
@@ -65,6 +69,35 @@ def test_list(self):
...
+            response = self.client.get(self.list_path)
+
+        self.assertEqual(response.status_code, 200)
+        self.assert_response_data_valid(response, Organization.objects.all())
+
+        # Check non staff user gets 1 group
+        self.client.logout()
+        self.client.login(username=self.non_staff_user.username, password=USER_PASSWORD)
+
+        with self.assertNumQueries(11):
```
Comment (jlajoie) [inline course_discovery/apps/api/v1/tests/test_views/test_organizations.py]: nit: same as above

**Review by jlajoie — APPROVED** [review]
Just nits about the numQueries tests, would rather we test for results rather than the means to get them (more/less queries don't feel like they should break tests imo)

File: course_discovery/apps/api/v1/tests/test_views/test_organizations.py (hunk 1 above)
Comment (staubina) [reply course_discovery/apps/api/v1/tests/test_views/test_organizations.py]: I almost removed them, they are remnants of the previous tests.
 
They did rise quickly when I stopped using the prefetch.

**Pull request merged**